bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -0,0 +1,443 @@
+        if (command.equals(RefreshType.REFRESH)) {
+            refresh();
+        } else {
+            logger.debug("Handling command {} for channel {}", command, channelUID);
+            String groupId = channelUID.getGroupId();
+            if (groupId == null)
+                return;
+            String channelIdWithoutGroup = channelUID.getIdWithoutGroup();
+            int loadpoint = Integer.parseInt(groupId.toString().substring(9));
+            EvccAPI evccAPI = this.evccAPI;
+            if (evccAPI == null)
+                return;
+            switch (channelIdWithoutGroup) {
+                case CHANNEL_LOADPOINT_MODE:
+                    evccAPI.setMode(loadpoint, command.toString());
+                    break;
+                case CHANNEL_LOADPOINT_MIN_SOC:
+                    evccAPI.setMinSoC(loadpoint, Integer.parseInt(command.toString().replaceAll(" %", "")));
+                    break;
+                case CHANNEL_LOADPOINT_TARGET_SOC:
+                    evccAPI.setTargetSoC(loadpoint, Integer.parseInt(command.toString().replaceAll(" %", "")));
+                    break;
+                case CHANNEL_LOADPOINT_TARGET_TIME:
+                    if (targetTimeEnabled == true) {
+                        targetTimeZDT = new DateTimeType(command.toString()).getZonedDateTime();
+                        evccAPI.setTargetCharge(loadpoint, targetSoC, targetTimeZDT);
+                        ChannelUID channel = new ChannelUID(getThing().getUID(), "loadpoint" + loadpoint,
+                                CHANNEL_LOADPOINT_TARGET_TIME);
+                        updateState(channel, new DateTimeType(targetTimeZDT));
+                    }
+                    break;
+                case CHANNEL_LOADPOINT_TARGET_TIME_ENABLED:
+                    if (command == OnOffType.ON) {
+                        targetTimeEnabled = true;
+                        evccAPI.setTargetCharge(loadpoint, targetSoC, targetTimeZDT);
+                    } else {
+                        targetTimeEnabled = false;
+                        evccAPI.unsetTargetCharge(loadpoint);
+                    }
+                    break;
+                case CHANNEL_LOADPOINT_PHASES:
+                    evccAPI.setPhases(loadpoint, Integer.parseInt(command.toString()));
+                    break;
+                case CHANNEL_LOADPOINT_MIN_CURRENT:
+                    evccAPI.setMinCurrent(loadpoint, Integer.parseInt(command.toString().replaceAll(" A", "")));
+                    break;
+                case CHANNEL_LOADPOINT_MAX_CURRENT:
+                    evccAPI.setMaxCurrent(loadpoint, Integer.parseInt(command.toString().replaceAll(" A", "")));
+                    break;
+                default:
+                    return;
+            }
+            refresh();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        this.config = getConfigAs(EvccConfiguration.class);
```

Comment:
`getConfigAs()` will not return null. And also it seems that the values obtained from the configuration is only used locally within `initialize()`. So this can be simplified by just validating URL and using the values directly without storing in any instance variables.